api/v1alpha1/clusterextension_types.go

```diff
@@ -61,12 +61,53 @@ type ClusterExtensionSpec struct {
 	//
 	Source SourceConfig `json:"source"`
 
-	// installNamespace is a reference to the Namespace in which the bundle of
+	// install is a required field used to configure the installation options
+	// for the ClusterExtension such as the installation namespace,
+	// the service account and the pre-flight check configuration.
+	//
+	// Below is a minimal example of an installation definition (in yaml):
+	// install:
+	//    namespace: example-namespace
+	//    serviceAccount:
+	//      name: example-sa
+	Install InstallationConfig `json:"install"`
```

Comment:
Let's try to be consistent with word forms between field names and type names:
```suggestion
	Install InstallConfig `json:"install"`
```

Reply:
Also, should we be prefixing these type names with `ClusterExtension`?

This package will contain structs for any future root API types and their sub-structs.

A hypothetical `Extension` API might have its own `InstallConfig` struct, but it will be different because it would not have a `namespace` field.

Reply:
Do you mean it should be `ClusterExtensionInstallConfig`?

Reply:
I think `ClusterExtensionInstallConfig` is reasonable

Reply:
But I think @joelanford is referring to all the new types here. So something like:
- `ClusterExtensionInstallConfig`
- `ClusterExtensionInstallStatus`
- `ClusterExtensionResolutionStatus`
- ...